src/utilities.ts

```diff
@@ -236,3 +244,91 @@ export function getInvalidAjvMessage(validator: any, input: any) {
     })
     .join(', ')}`;
 }
+
+export function generateAuctionPermutations(
+  props: AuctionGeneratorProps,
+): AuctionPermutations {
+  const auctions = {};
+  for (let i = 0; i <= MIN_YEARS; i++) {
+    auctions[i] = generateAuction({
+      ...props,
+      years: i ?? undefined, // 0 = permabuy-
+    });
+  }
+
+  return auctions as AuctionPermutations;
+}
+
+export function generateAuction(props: AuctionGeneratorProps): Auction {
+  // calculate the auctions prices for each year (leases) and for permabuy
+
+  const {
+    name,
+    blockHeight,
+    blockTime,
+    fees,
+    auctionSettings,
+    caller,
+    years,
+    bid,
+    contractTxId,
+    type,
+  } = props;
+
+  const { floorPriceMultiplier, startPriceMultiplier, id } = auctionSettings;
+
+  const registrationFee = years
+    ? calculateTotalRegistrationFee(name, fees, years!, blockTime)
+    : calculatePermabuyFee(name, fees, blockTime);
+
```

Comment:
i'd like to see specific unit tests for this helper function

Reply:
added